### ISSUE TITLE
Constant if expression?

### ISSUE THREAD

**Issue by dharkness** [bitpressio/blade-extensions#2]
I'd like to use this to alter the template output when it's being compiled based on a `.env` setting that's fixed at deployment. We have two news sites running on the same codebase, and a few pages are drastically altered by which site is being served. We compile the templates during deployment, and I'd like to fix those alterations in the compiled output so that the `if` condition is not checked during runtime. Do you know if this is possible?

In other words, I'd like to tell the Blade compiler to evaluate the `if` condition during compilation instead of injecting `<php if ($constant_condition): ?>` into the compiled template. I suspect I'll have to extend the compiler itself, but I'm hoping you've already thought of that. :)

**Comment by dharkness** [comment]
After digging through the `BladeCompiler` source, I can see this would not be doable without a large refactoring. The parser would need conditional-compilation to omit the parts of the template to cut out, and saving a simple runtime check of `@if (config('app.site') === 'SITE1')` wouldn't be worth the effort.

**Comment by paulredmond** [comment]
Hey @dharkness, ya, this is just some automation to register classes as directives and conditionals and using the service container to inject any dependencies your methods need. Thanks for stopping by!

**Comment by dharkness** [comment]
I got excited when I saw `getConditionals()`, but your code ended up making overriding some compiler methods for a different feature much easier. Thanks! :)
